src/irmin/tree_intf.ml

```diff
@@ -76,6 +76,24 @@ module type S = sig
 
   (** {1 Manipulating Contents} *)
 
+  type 'a or_error = ('a, [ `Dangling_hash of hash ]) result
+  (** Operations on lazy nodes can fail if the underlying store does not contain
+      the expected hash. *)
+
+  (** Operations on lazy tree contents. *)
+  module Contents : sig
+    type t
+    (** The type of lazy tree contents. *)
+
+    val to_hash : t -> hash
```

Comment:
`hash` is probably better then `to_hash` (as you did initially). The function name in the comment is out-of-date.

Reply:
Unfortunately, `hash` is already being used internally... I'll do a pass to rename everything.

Reply:
Now renamed. The internal functions that check the current state of the lazy view are now called `cached_{value,hash,map}`, with exported `hash` functions that coerce the representation instead.